include/mlio/image_reader.h

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You
+ * may not use this file except in compliance with the License. A copy of
+ * the License is located at
+ *
+ *      http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+ * ANY KIND, either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ */
+
+#pragma once
+
+#include <opencv4/opencv2/core/mat.hpp>
+
+#include "mlio/config.h"
+#include "mlio/parallel_data_reader.h"
+
+namespace mlio {
+inline namespace v1 {
+
+/// @addtogroup data_readers Data Readers
+/// @{
+
+enum class image_frame { file, pipe };
+
+struct MLIO_API image_reader_params final {
+    /// Enum to identify 'file' or 'pipe' mode. Defaults to file.
+    image_frame img_frame{image_frame::file};
+    /// Down scale the shorter edge to a new size before applying other
+    /// augmentations. Defaults to -1 to disable resize.
+    int resize = -1;
```

Comment:
I assume this is a temporary option. When we infer the image dimensions from the dataset, there is no way for us to know what is the "shorter" edge. And why the shorter edge? Why not the longer edge? For resizing we should think of a more expressive API in the future.

Reply:
I don't like sentinel values. Please use `std::optional<std::size_t>` for this parameter.

Reply:
For (1): I was trying to keep the logic consistent with mxnet. They define resize operations has resizing the shorter edge, as mentioned here: https://mxnet.apache.org/api/python/docs/api/mxnet/io/index.html#mxnet.io.ImageRecordIter

Reply:
Again, it is okay for now. Going forward we should come up with a better API than what MXNet provides.